articles/151_roslaunch_xml.md

```diff
@@ -0,0 +1,744 @@
+      </xs:attribute>
+      <xs:attribute name="if" type="xs:string" use="optional">
+        <xs:annotation>
+          <xs:documentation xml:lang="en">
+            A predicate to condition ROS node launch i.e. only do
+            so if the predicate evaluates to a truthy value.
+          </xs:documentation>
+        </xs:annotation>
+      </xs:attribute>
+      <xs:attribute name="unless" type="xs:string" use="optional">
+        <xs:annotation>
+          <xs:documentation xml:lang="en">
+            A predicate to condition ROS node launch i.e. do so unless
+            the predicate evaluates to a truthy value.
+          </xs:documentation>
+        </xs:annotation>
+      </xs:attribute>
+  </xs:element>
+</xs:schema> 
+```
+
+### Tags Semantics
+
+#### All Tags
+
+Every tag's execution can be conditioned on a boolean predicate via `if` or `unless` attributes.
+
+#### <launch> Tag
+
+Root tag of any launch file.
+
+#### <include> Tag
+
+##### Description
+
+The <include> tag allows for bringing a launch file description into another, enabling re-use of hierarchical launch layouts.
+The included launch file description has its own scope for launch configurations.
+The included launch file description is not necessarily written in this format nor a declarative one (i.e. a programmatic description).
+To avoid ROS name clashes, included nodes can be namespaced via the `ns` attribute.
+
+##### Examples
+
+```xml
+<include file="/opt/my_launch_file.py" ns="/my_ns"/>
+<include file="/opt/my_other_launch_file.xml">
+  <arg name="some_argument" value="dummy_value"/>
+</include>
+```
+
+#### <group> Tag
+
+##### Description
+
+The <group> tag allows for launch actions' grouping as well as optional launch file configuration scoping.
+
+##### Examples
+
+```xml
+<group ns="/dummy_group" scoped="true">
+  <node package="a_ros_package" name="dummy0" executable="dummy_node"/>
+  <node package="a_ros_package" name="dummy1" executable="dummy_node"/>
+</group>
+```
+
+#### <arg> Tag
+
+##### Description
+
+The <arg> tag allows for launch file configuration via the command-line or when including it via an <include> tag.
+Arguments are limited to the scope of their definition and thus, have to be explictly passed to included files if any.
+
+##### Examples
+
+```xml
+<arg name="publish_frequency" default="10"/>
+<arg name="output_path" doc="Output path for some processing pipeline"/>
+```
+
+#### <node> Tag
+
+##### Description
+
+The <node> tag allows for executing a ROS node in the form of local OS process.
+
+##### Examples
+
+```xml
+<node package="ros_demos" executable="publisher">
+  <param name="publish_frequency">10</param>
+  <remap from="generic_topic_name" to="my_topic"/>
+</node>
+```
+
+#### <param> Tag
+
+##### Description
+
+The <param> tag allows for setting a ROS parameter of a ROS node.
+
+##### Examples
+
+```xml
+<param name="some_numeric_param" value="100.2"/>
+<param name="some_list_param" value="Some phrase,100.0,true" sep=","/>
```

Comment:
It does need to be inside of a `<node>` tag, which is maybe implied, but it would be good to have in the example as well. In ROS 1, `<param>` could be outside of a node tag and in that case would set global parameters. We might have something like that for ROS 2 in the future, but right now it would not do anything.

Reply:
Same comment for `<params>` below.

Reply:
Makes sense. Enclosed them in `<node>` tags.